torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -580,28 +582,37 @@ static void eraseListConstruct(Block* block) {
   }
 }
 
-static void fuseSplitListUnpack(Block* b) {
+static void fuseListUnpack(Block* b) {
   for (auto it = b->nodes().begin(), end = b->nodes().end(); it != end; ++it) {
+
     for (auto* child_block : it->blocks()) {
-      fuseSplitListUnpack(child_block);
-    }
-    if (it->kind() == prim::ListUnpack &&
-        it->input()->node()->kind() == onnx::Split) {
-      auto origSplitNode = it->input()->node();
+      fuseListUnpack(child_block);
+    }
+    if (it->kind() == prim::ListUnpack) {
+      Node* input_node = it->input()->node();
+      if (input_node->kind() == prim::ListConstruct) {
```

Comment:
We should document why we can have such weired behavior ListConstruct + ListUnpack together. 

Also I prefer to have a separate pass like https://github.com/pytorch/pytorch/pull/27247/files#diff-4043567f1b468c464247ad597729d6c0R624

Reply:
@BowenBao I know you are working on a fix for split. Do we still need this pass after your fix is done?

Reply:
I added notes. This pass is a temporary one, and in any case I'm looking into refactoring the parts where this handles list unpack as well. Let's have this fix for meshgrid first, and we'll refactor the code in a new PR.

Reply:
@houseroad I've fixed the pass